### ISSUE TITLE
Unable to add integrations in pnpm turborepo

### ISSUE THREAD

**Issue by atanaskanchev** [withastro/astro#8487]
### Astro Info

```block
Astro                    v3.0.12
Node                     v18.16.1
System                   macOS (arm64)
Package Manager          pnpm
Output                   static
Adapter                  none
Integrations             none
```

### Describe the Bug

1. `pnpm dlx create-turbo@latest`
2. `pnpm create astro@latest`
```
 astro   v3.0.12 Launch sequence initiated.

   dir   Where should we create your new project?
         ./apps/web

  tmpl   How would you like to start your new project?
         Empty
      ✔  Template copied

  deps   Install dependencies?
         No
      ◼  No problem! Remember to install dependencies after setup.

    ts   Do you plan to write TypeScript?
         Yes

   use   How strict should TypeScript be?
         Strictest
      ✔  TypeScript customized

   git   Initialize a new git repository?
         No
      ◼  Sounds good! You can always run git init manually.

  next   Liftoff confirmed. Explore your project!
```
3. `cd apps/web`
```
  web git:(main) ✗ pnpm astro add react

> apps-web@0.0.1 astro /Users/atanaskanchev/Downloads/astro-integrations-turborepo-pnpm/apps/web
> astro "add" "react"

✔ Resolving packages...

  Astro will run the following command:
  If you skip this step, you can always run it yourself later

 ╭─────────────────────────────────────────────────────────────────────────────────────────────────────────╮
 │ pnpm add @astrojs/react @types/react-dom@^18.0.6 @types/react@^18.0.21 react-dom@^18.0.0 react@^18.0.0  │
 ╰─────────────────────────────────────────────────────────────────────────────────────────────────────────╯

✔ Continue? … yes
✖ Installing dependencies...

  ERR_PNPM_INCLUDED_DEPS_CONFLICT  modules directory (at "/Users/atanaskanchev/Downloads/astro-integrations-turborepo-pnpm") was installed with optionalDependencies, dependencies, devDependencies. Current install wants optionalDependencies, dependencies. 

 error   Astro could not update your astro.config.js file safely.
  Reason: Unable to install dependencies
  
  You will need to add these integration(s) manually.
  Documentation: https://docs.astro.build/en/guides/integrations-guide/
  File:
    /Users/atanaskanchev/Downloads/astro-integrations-turborepo-pnpm/node_modules/.pnpm/astro@3.0.12/node_modules/astro/dist/cli/add/index.js:169:31
  Code:
    168 |     case 3 /* failure */: {
    > 169 |       throw createPrettyError(new Error(`Unable to install dependencies`));
          |                               ^
      170 |     }
      171 |   }
      172 |   const rawConfigPath = await resolveConfigPath({
  Stacktrace:
Error: Astro could not update your astro.config.js file safely.
Reason: Unable to install dependencies

You will need to add these integration(s) manually.
Documentation: https://docs.astro.build/en/guides/integrations-guide/
    at add (file:///Users/atanaskanchev/Downloads/astro-integrations-turborepo-pnpm/node_modules/.pnpm/astro@3.0.12/node_modules/astro/dist/cli/add/index.js:169:31)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async runCommand (file:///Users/atanaskanchev/Downloads/astro-integrations-turborepo-pnpm/node_modules/.pnpm/astro@3.0.12/node_modules/astro/dist/cli/index.js:102:7)
    at async cli (file:///Users/atanaskanchev/Downloads/astro-integrations-turborepo-pnpm/node_modules/.pnpm/astro@3.0.12/node_modules/astro/dist/cli/index.js:144:5)

 ELIFECYCLE  Command failed with exit code 1.
```

### What's the expected result?

Should be able to install integrations in turborepo app using pnpm

### Link to Minimal Reproducible Example

https://github.com/atanaskanchev/astro-integrations-turborepo-pnpm.git

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by ematipico** [comment]
Have you tried to install the dependencies when running create astro? Is there a reason why you decided not to do so? 

This may be a legitimate error because `@astrojs/react` requires astro installed, but you haven't installed it.

**Comment by atanaskanchev** [comment]
> Have you tried to install the dependencies when running create astro? Is there a reason why you decided not to do so?
> 
> This may be a legitimate error because `@astrojs/react` requires astro installed, but you haven't installed it.

I'm sorry, I do not quite follow. So I first create the app with `pnpm create astro@latest`. At this point astro dependency is installed. Then I'm using it to add the react integration with `pnpm astro add react`. Is there a step I am missing?

**Comment by ematipico** [comment]
Looking at the prompt that you pasted, you **didn't** install the dependencies:

>  deps   Install dependencies?
         No
      ◼  No problem! Remember to install dependencies after setup.

**Comment by atanaskanchev** [comment]
Hi @ematipico unfortunately the same happens regardless of the choice in the `Install dependencies?` step

```
➜  astro-integrations-turborepo-pnpm git:(main) ✗ pnpm create astro@latest
.../Library/pnpm/store/v3/tmp/dlx-84374  | Progress: resolved 1, reused 0, downlo.../Library/pnpm/store/v3/tmp/dlx-84374  | Progress: resolved 20, reused 15, down.../Library/pnpm/store/v3/tmp/dlx-84374  |  +57 ++++++
.../Library/pnpm/store/v3/tmp/dlx-84374  | Progress: resolved 20, reused 15, down.../Library/pnpm/store/v3/tmp/dlx-84374  | Progress: resolved 57, reused 57, down.../Library/pnpm/store/v3/tmp/dlx-84374  | Progress: resolved 57, reused 57, downloaded 0, added 57, done

╭─────╮  Houston:
│ ◠ ◡ ◠  Awaiting further instructions.
╰─────╯

 astro   v3.0.12 Launch sequence initiated.

   dir   Where should we create your new project?
         ./apps/web

  tmpl   How would you like to start your new project?
         Include sample files
      ✔  Template copied

  deps   Install dependencies?
         Yes
      ✔  Dependencies installed

    ts   Do you plan to write TypeScript?
         Yes

   use   How strict should TypeScript be?
         Strict
      ✔  TypeScript customized

   git   Initialize a new git repository?
         No
      ◼  Sounds good! You can always run git init manually.

  next   Liftoff confirmed. Explore your project!

         Enter your project directory using cd ./apps/web 
         Run pnpm dev to start the dev server. CTRL+C to stop.
         Add frameworks like react or tailwind using astro add.

         Stuck? Join us at https://astro.build/chat

╭─────╮  Houston:
│ ◠ ◡ ◠  Good luck out there, astronaut! 🚀
╰─────╯
➜  astro-integrations-turborepo-pnpm git:(main) ✗ cd apps/web    
➜  web git:(main) ✗ pnpm astro add react

> apps-web@0.0.1 astro /Users/atanaskanchev/Downloads/astro-integrations-turborepo-pnpm/apps/web
> astro "add" "react"

✔ Resolving packages...

  Astro will run the following command:
  If you skip this step, you can always run it yourself later

 ╭─────────────────────────────────────────────────────────────────────────────────────────────────────────╮
 │ pnpm add @astrojs/react @types/react-dom@^18.0.6 @types/react@^18.0.21 react-dom@^18.0.0 react@^18.0.0  │
 ╰─────────────────────────────────────────────────────────────────────────────────────────────────────────╯

✔ Continue? … yes
✖ Installing dependencies...

  ERR_PNPM_INCLUDED_DEPS_CONFLICT  modules directory (at "/Users/atanaskanchev/Downloads/astro-integrations-turborepo-pnpm") was installed with optionalDependencies, dependencies, devDependencies. Current install wants optionalDependencies, dependencies. 

 error   Astro could not update your astro.config.js file safely.
  Reason: Unable to install dependencies
  
  You will need to add these integration(s) manually.
  Documentation: https://docs.astro.build/en/guides/integrations-guide/
  File:
    /Users/atanaskanchev/Downloads/astro-integrations-turborepo-pnpm/node_modules/.pnpm/astro@3.0.12/node_modules/astro/dist/cli/add/index.js:169:31
  Code:
    168 |     case 3 /* failure */: {
    > 169 |       throw createPrettyError(new Error(`Unable to install dependencies`));
          |                               ^
      170 |     }
      171 |   }
      172 |   const rawConfigPath = await resolveConfigPath({
  Stacktrace:
Error: Astro could not update your astro.config.js file safely.
Reason: Unable to install dependencies

You will need to add these integration(s) manually.
Documentation: https://docs.astro.build/en/guides/integrations-guide/
    at add (file:///Users/atanaskanchev/Downloads/astro-integrations-turborepo-pnpm/node_modules/.pnpm/astro@3.0.12/node_modules/astro/dist/cli/add/index.js:169:31)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async runCommand (file:///Users/atanaskanchev/Downloads/astro-integrations-turborepo-pnpm/node_modules/.pnpm/astro@3.0.12/node_modules/astro/dist/cli/index.js:102:7)
    at async cli (file:///Users/atanaskanchev/Downloads/astro-integrations-turborepo-pnpm/node_modules/.pnpm/astro@3.0.12/node_modules/astro/dist/cli/index.js:144:5)

 ELIFECYCLE  Command failed with exit code 1.
```

**Comment by millette** [comment]
I use verdaccio (local npm registry) and pnpm and it fails for me too.

```sh
pnpm astro add tailwind

> waglo@0.0.1 astro /home/millette/travail/waglo
> astro "add" "tailwind"

✖ @astrojs/tailwind doesn't appear to be an integration or an adapter. 
Find our official integrations at https://astro.build/integrations
ELIFECYCLE  Command failed with exit code 1.
```

Same error for all integrations I try adding.

**Comment by michelemadeddu** [comment]
Hi, same ERR_PNPM_INCLUDED_DEPS_CONFLICT  error installing react integration through "pnpm astro add react", though I'm not using turborepo in my monorepo.

**Comment by florian-lefebvre** [comment]
Facing same issue on a simple pnpm monorepo

**Comment by dr7v3** [comment]
I've had this issue as well for the last couple of months. I've recently updated node (v21.2.0), pnpm (8.10.5), and running the lastest version of astro (3.5.4). Did not have an issue this time around, I also added the latest @turbo/gen as a devDependencie to the package.json of turborepo, something like this:

"scripts": {
    "build": "turbo run build",
    "dev": "turbo run dev"
  },
"devDependencies": {
    "@turbo/gen": "^1.10.16",
    "turbo": "latest"
  },
  "packageManager": "pnpm@8.10.5",
  "name": "turbo"

Hope this helps!
